src/zos_files/zowe/zos_files_for_zowe_sdk/files.py

```diff
@@ -122,6 +122,20 @@ def get_dsn_content(self, dataset_name):
         response_json = self.request_handler.perform_request("GET", custom_args)
         return response_json
 
+    def get_dsn_content_streamed(self, dataset_name):
+        """Retrieve the contents of a given dataset streamed.
+
+        Returns
+        -------
+        json
+            A JSON with the contents of a given dataset
+        """
+        custom_args = self.create_custom_request_arguments()
+        custom_args["url"] = "{}ds/{}".format(self.request_endpoint, dataset_name)
+        custom_args["stream"] = True
```

Comment:
What about adding this line to `request_handler.normalize_response`
```python
if 'stream' in self.request_arguments and self.request_arguments['stream'] == True:
    return self.response.raw
```
Then in `download_dsn`
```python
raw_response = self.get_dsn_content_streamed(dataset_name)
out_file = open(output_file, 'w')
shutil.copyfileobj(raw_response, out_file)
out_file.close()

Reply:
Good idea for `request_handler.normalize_response`, I think that will be nicer than creating a separate `_streamed` method 👍 

In `download_dsn` instead of calling `out_file.close`, I'd suggest using a `with` block:
```
with open(output_file, 'w') as out_file:
    shutil.copyfileobj(raw_response, out_file)
```